kong/router/traditional.lua

```diff
@@ -1448,6 +1450,21 @@ function _M.new(routes, cache, cache_neg)
   local match_sources        = not isempty(plain_indexes.sources)
   local match_destinations   = not isempty(plain_indexes.destinations)
 
+  -- warning about the regex cache size being too small
+  if not lua_regex_cache_max_entries then
+    lua_regex_cache_max_entries = kong.configuration.nginx_http_lua_regex_cache_max_entries
+                              and tonumber(kong.configuration.nginx_http_lua_regex_cache_max_entries) or 1024
+  end
+
+  if worker_id() == 0 and regex_uris[0] > lua_regex_cache_max_entries then
```

Comment:
I think we should also be checking for `regex_uris[0] * 2 > lua_regex_cache_max_entries` here, if user sets `lua_regex_cache_max_entries` to be exactly `regex_uris[0]` we can still expect issues regarding regex cache trashing.

Reply:
Thanks, done.